contracts/router/Router.sol

```diff
@@ -0,0 +1,370 @@
+
+        7. Call swapAndBridge(
+            _amountIn,
+            _minAmountOut,
+            _path,
+            _adapters,
+            _bridgeData
+        )
+
+        8. Use swapFromGasAndBridge with same params instead, 
+           if you want to start from GAS
+     */
+
+    /// @dev Use this function, when doing a "swap into ETH and bridge" on Mainnet,
+    /// as bridging from ETH Mainnet requires depositing WETH into bridge contract
+    /// This is why there's so "swapToGasAndBridge()" implemented
+    /// The same applies to "swap into BNB and bridge" on BNB,
+    /// "swap into AVAX and bridge" on Avalanche, etc
+    function swapAndBridge(
+        uint256 _amountIn,
+        uint256 _minAmountOut,
+        address[] calldata _path,
+        address[] calldata _adapters,
+        bytes calldata _bridgeData
+    ) external {
+        uint256 _swapAmount = _swap(
+            _amountIn,
+            _minAmountOut,
+            _path,
+            _adapters,
+            msg.sender,
+            address(this)
+        );
+        _callBridge(_path[_path.length - 1], _swapAmount, _bridgeData);
+    }
+
+    function swapFromGasAndBridge(
+        uint256 _amountIn,
+        uint256 _minAmountOut,
+        address[] calldata _path,
+        address[] calldata _adapters,
+        bytes calldata _bridgeData
+    ) external payable {
+        require(msg.value == _amountIn, "Router: incorrect amount of GAS");
+        require(_path[0] == WGAS, "Router: path needs to begin with WGAS");
+        _wrap(_amountIn);
+        // WGAS tokens need to be sent from this contract
+        uint256 _swapAmount = _selfSwap(
+            _amountIn,
+            _minAmountOut,
+            _path,
+            _adapters,
+            address(this)
+        );
+        _callBridge(_path[_path.length - 1], _swapAmount, _bridgeData);
+    }
+
+    function _callBridge(
+        address _bridgeToken,
+        uint256 _bridgeAmount,
+        bytes calldata _bridgeData
+    ) internal {
+        _setBridgeTokenAllowance(_bridgeToken, _bridgeAmount);
+        (bool success, ) = bridge.call(_bridgeData);
+        require(success, "Bridge interaction failed");
+    }
+
+    // -- BRIDGE RELATED FUNCTIONS [destination chain] --
+
+    /** @dev
+        Bridge contract is supposed to 
+        1. Transfer tokens (token: _path[0]; amount: _amountIn) to Router contract
+
+        2. Call ROUTER.selfSwap(...)
+
+        3. If swap succeeds, no need to do anything, tokens are at _to address
+                If _path ends with WGAS, user will receive GAS instead of WGAS
+
+        4. If selfSwap() reverts, bridge is supposed to call 
+                refundToAddress(_path[0], _amountIn, _to);
+            This will return intermediate token (nUSD, nETH, ...) to the user
+     */
+
+    function refundToAddress(
+        address _token,
+        uint256 _amount,
+        address _to
+    ) external onlyBridge {
```

Comment:
While this is privileged to the bridge, don't see downside to adding re-entrancy guard given low gas cost.

Reply:
Fun(?) fact: even though `_token` is a bridged token, it can be GAS, if you bridge GAS back to its native chain & have an unsuccessful swap.

Fixed in 766a739